Incorrect / incomplete documentation regarding junit-platform-launcher incompatibilities

**Describe the bug**
The documentation at https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#testing-support-version-compatibility currently says that for old Surefire plugin versions it is necessary to add an explicit `junit-platform-launcher` dependency, but for newer versions this is not necessary.

However, this information might be incorrect or incomplete. It seems this Native Maven Plugin has itself a dependency on `junit-platform-launcher` and is accessing its classes through `org.graalvm.junit.platform.JUnitPlatformFeature#discoverTestsAndRegisterTestClassesForReflection`, without any involvement of Surefire (?).

For example it seems with JUnit 5.12.0 and native-maven-plugin 0.10.5 and 0.10.6 the step "Running in 'test listener' mode ..." fails with:

```
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [...\gson\test-graal-native-image\target\test-ids] and its subfolders.

Error: Feature defined by org.graalvm.junit.platform.JUnitPlatformFeature unexpectedly failed with a(n) org.junit.platform.commons.JUnitException. Please report this problem to the authors of org.graalvm.junit.platform.JUnitPlatformFeature.
com.oracle.svm.core.util.UserError$UserException: Feature defined by org.graalvm.junit.platform.JUnitPlatformFeature unexpectedly failed with a(n) org.junit.platform.commons.JUnitException. Please report this problem to the authors of org.graalvm.junit.platform.JUnitPlatformFeature.
        at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.UserError.abort(UserError.java:87)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.handleFeatureError(FeatureHandler.java:293)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:92)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:774)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:593)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:551)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:539)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:721)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.start(NativeImageGeneratorRunner.java:143)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:98)
Caused by: org.junit.platform.commons.JUnitException: TestEngine with ID 'junit-jupiter' failed to discover tests
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:160)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverSafely(EngineDiscoveryOrchestrator.java:132)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:107)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discover(EngineDiscoveryOrchestrator.java:78)
        at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:99)
        at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:77)
        at org.junit.platform.launcher.core.DelegatingLauncher.discover(DelegatingLauncher.java:42)
        at org.junit.platform.launcher.core.SessionPerRequestLauncher.discover(SessionPerRequestLauncher.java:56)
        at org.graalvm.junit.platform.JUnitPlatformFeature.discoverTestsAndRegisterTestClassesForReflection(JUnitPlatformFeature.java:135)
        at org.graalvm.junit.platform.JUnitPlatformFeature.beforeAnalysis(JUnitPlatformFeature.java:94)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.lambda$runPointsToAnalysis$9(NativeImageGenerator.java:774)
        at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.FeatureHandler.forEachFeature(FeatureHandler.java:90)
        ... 7 more
Caused by: org.junit.platform.commons.JUnitException: OutputDirectoryProvider not available; probably due to unaligned versions of the junit-platform-engine and junit-platform-launcher jars on the classpath/module path.
        at org.junit.platform.engine.EngineDiscoveryRequest.getOutputDirectoryProvider(EngineDiscoveryRequest.java:94)
        at org.junit.jupiter.engine.JupiterTestEngine.discover(JupiterTestEngine.java:67)
        at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:152)
        ... 18 more
```
(requires `-H:+ReportExceptionStackTraces` to see the stack trace, see https://github.com/graalvm/native-build-tools/issues/541#issuecomment-2727121186)

**To Reproduce**

1. Clone https://github.com/google/gson/pull/2818/commits/832382f70aadf65cf772570d6021ee6b0a622247
2. Set up GraalVM on `PATH` and `JAVA_HOME`
3. Run
    ```bash
    mvn test --activate-profiles native-image-test --projects test-graal-native-image --also-make
    ```

If you want I can also try to create a simplified example.

**Expected behavior**
- Ideally the `JUnitException` should not occur in the first place
The Native Maven Plugin should on its own determine the necessary `junit-platform-launcher` version.
- If that is not possible, please adjust the documentation to mention that it might also be necessary to add a `junit-platform-launcher` dependency to solve incompatibilities with the Native Maven Plugin, and not only Surefire
- Consider enabling `-H:+ReportExceptionStackTraces` by default
Without stack traces troubleshooting the original error message is impossible, and finding out about the `-H:+ReportExceptionStackTraces` argument might also not be easy if it is not clear to the user what and where to look for.

**Logs**
_see stack trace above_

**System Info (please complete the following information):**
 - OS: Windows 10
 - GraalVM Version: `21.0 CE`
 - Java Version: 21 (`Oracle GraalVM 21.0.6+8.1`)
 - Plugin version: `native-maven-plugin:0.10.6`

**Additional context**
_none_

--- a/docs/src/docs/asciidoc/maven-plugin.adoc
+++ b/docs/src/docs/asciidoc/maven-plugin.adoc
@@ -675,7 +675,6 @@ Running the `./mvnw -Pnative test` will then build and run tests as native code.
 
 This plugin requires the Maven Surefire plugin 3.0.0 or higher to run tests as native code.
 
-// Native test classpath and launcher/feature behavior are specified by §maven/FS-maven-native-tests.1 and §maven/FS-maven-native-tests.3.
 Native test execution also uses the JUnit Platform while building the native test image.
 By default, the plugin builds the test image with `NativeImageJUnitLauncher` and `JUnitPlatformFeature`,
 which discover and register selected tests using the JUnit Platform artifacts on the test classpath.
@@ -720,7 +719,6 @@ If Maven Surefire is using an older version of the JUnit Platform, the build wil
 [ERROR] Test configuration file wasn't found. Make sure that test execution wasn't skipped.
 ----
 
-// Native-image analysis registration is specified by §FS-native-tests.2.
 If the native test classpath contains incompatible JUnit Platform launcher and engine versions,
 native-image analysis can also fail during test discovery with an error similar to the following:
 

--- a/docs/src/docs/asciidoc/maven-plugin.adoc
+++ b/docs/src/docs/asciidoc/maven-plugin.adoc
@@ -675,11 +675,25 @@ Running the `./mvnw -Pnative test` will then build and run tests as native code.
 
 This plugin requires the Maven Surefire plugin 3.0.0 or higher to run tests as native code.
 
+// Native test classpath and launcher/feature behavior are specified by §maven/FS-maven-native-tests.1 and §maven/FS-maven-native-tests.3.
+Native test execution also uses the JUnit Platform while building the native test image.
+By default, the plugin builds the test image with `NativeImageJUnitLauncher` and `JUnitPlatformFeature`,
+which discover and register selected tests using the JUnit Platform artifacts on the test classpath.
+For that reason, the `junit-platform-launcher` artifact available to native test execution must be
+compatible with the JUnit Platform engine artifacts, such as JUnit Jupiter or JUnit Vintage.
+
 Each version of the Maven Surefire plugin has a dependency on a particular version of the JUnit Platform.
-You therefore need to ensure that Maven Surefire is using at least version 1.8 of the JUnit Platform.
-Beginning with version 3.0 M4, Maven Surefire automatically aligns the JUnit Platform version used by Surefire with the version needed by the user's configured version of JUnit Jupiter or JUnit Vintage.
-Since the examples here use Maven Surefire 3.0 M5, their maven POMs do not require any special configuration regarding the JUnit Platform version.
-However, if you are using a version of Maven Surefire prior to 3.0 M4, you will need to add an explicit dependency on the `junit-platform-launcher` artifact to the `dependencies` section of your `native` profile configuration. For example:
+Ensure that Maven Surefire and Native Build Tools use compatible JUnit Platform artifacts,
+and that those artifacts are at least version 1.8.
+Beginning with version 3.0 M4, Maven Surefire automatically aligns the JUnit Platform version used by
+Surefire with the version needed by the user's configured version of JUnit Jupiter or JUnit Vintage.
+Since the examples here use Maven Surefire 3.0 M5, their Maven POMs do not require any special configuration regarding the JUnit Platform version.
+However, if you are using a version of Maven Surefire prior to 3.0 M4, or if your native test
+classpath otherwise contains mismatched JUnit Platform artifacts, you will need to add an explicit
+dependency on the `junit-platform-launcher` artifact to the `dependencies` section of your `native`
+profile configuration.
+Choose a `junit-platform-launcher` version managed by your JUnit BOM or one that matches the JUnit Platform version used by your test engines.
+For example:
 
 [source,xml, role="multi-language-sample"]
 ----
@@ -704,6 +718,15 @@ If Maven Surefire is using an older version of the JUnit Platform, the build wil
 
 ----
 [ERROR] Test configuration file wasn't found. Make sure that test execution wasn't skipped.
+----
+
+// Native-image analysis registration is specified by §FS-native-tests.2.
+If the native test classpath contains incompatible JUnit Platform launcher and engine versions,
+native-image analysis can also fail during test discovery with an error similar to the following:
+
+----
+OutputDirectoryProvider not available; probably due to unaligned versions of the junit-platform-engine
+and junit-platform-launcher jars on the classpath/module path.
 ----
 
 [[testing-support-disabling]]

